sui_programmability/examples/games/tests/vdf_based_lottery_tests.move

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+#[test_only]
+module games::vdf_based_lottery_tests {
+    use sui::test_scenario::{Self};
+    use games::vdf_based_lottery::{Self, Game, Ticket, GameWinner};
+
+    #[test]
+    #[expected_failure(abort_code = games::vdf_based_lottery::ESubmissionPhaseInProgress)]
+    fun test_complete_too_early() {
+        let user1 = @0x0;
+
+        let scenario_val = test_scenario::begin(user1);
+        let scenario = &mut scenario_val;
```

Comment:
Because of receiver syntax, you can now avoid the pattern of creating a variable to hold the mutable reference for the scenario, just do:

```move
let ts = ts::begin(user1);
let clock = clock::create_for_testing(ts.ctx());
```

(assumes imports for sui::test_scenario as ts and sui::clock).